model-optimizer/mo/back/preprocessing.py

```diff
@@ -0,0 +1,306 @@
+                    .format(name))
+
+    dim_idx_found = -1
+    for dim_idx in range(shape.rank.get_length()):
+        dim = shape.get_dimension(dim_idx)
+        if dim.is_static and dim.get_length() == num_channels:
+            if dim_idx_found >= 0:
+                raise Error('Can\'t determine channels dimension for {}. '
+                            'Input shape is {}, needed channels {}. '
+                            'Conflicting dimensions: {} and {}'
+                            .format(name, shape, num_channels, dim_idx_found, dim_idx))
+            dim_idx_found = dim_idx
+    if dim_idx_found < 0:
+        raise Error('Can\'t determine channels dimension for {}. '
+                    'Input shape is {}, needed channels {}'
+                    .format(name, shape, num_channels))
+    layout_str = "?" * shape.rank.get_length()
+    layout_str = layout_str[:dim_idx_found] + 'C' + layout_str[dim_idx_found+1:]
+    layout_values[name] = {
+        'source_layout': layout_str,
+        'target_layout': None,
+        'source_guessed': True,
+        'is_input': True
+    }
+    return layout_values
+
+
+def guess_source_layouts_by_mean_scale(ov_function: Function, layout_values, mean_scale_values: dict):
+    """
+    Internal function. Try to guess source layout for input by its shape and/or framework
+    :param: ov_function Original model
+    :param: layout_values Existing source/target layout items specified by user
+    :param: mean_scale_values Dictionary with mean/scale values defined for each argument
+    :return: updated layout items with guessed layouts
+    """
+    for ms_name, mean_scale in mean_scale_values.items():
+        num_channels_mean = len(mean_scale['mean']) if mean_scale['mean'] is not None else 0
+        num_channels_scale = len(mean_scale['scale']) if hasattr(mean_scale['scale'], '__len__') else 0
+
+        if num_channels_mean > 1 and \
+                num_channels_scale > 1 and \
+                num_channels_mean is not num_channels_scale:
+            raise Error('Mean/Scale values for {} have different sizes: {} {}'
+                        .format(ms_name, num_channels_mean, num_channels_scale))
+
+        need_guess_channels = num_channels_mean > 1 or num_channels_scale > 1
+        if not need_guess_channels:  # Mean/scale is complex and needs 'channels' specified in layout
+            continue
+
+        num_channels = num_channels_mean if num_channels_mean > 1 else num_channels_scale
+        for ov_input in ov_function.inputs:
+            if ms_name not in ov_input.get_tensor().get_names():
+                continue
+
+            layout_exists = False
+            layout_item = None
+            for name in ov_input.get_tensor().get_names():
+                if name in layout_values:
+                    layout_item = layout_values[name]
+                    if layout_item.get('source_layout'):
+                        layout_exists = True
```

Comment:
Also need to check if 'layout' parameter's runtime info exists - skip guessing layout. In future, frontends may set such layout during conversion

Reply:
Fixed, tests added